src/pages/spawn/spawnHost/SpawnHostActionButton.tsx

```diff
@@ -1,23 +1,110 @@
-import React from "react";
+import React, { useEffect } from "react";
+import { useMutation, useLazyQuery } from "@apollo/client";
 import styled from "@emotion/styled";
 import Button, { Size } from "@leafygreen-ui/button";
 import Icon from "components/icons/Icon";
+import { useBannerDispatchContext } from "context/banners";
+import {
+  UpdateSpawnHostStatusMutation,
+  UpdateSpawnHostStatusMutationVariables,
+  SpawnHostStatusActions,
+  MyHostsQuery,
+  MyHostsQueryVariables,
+} from "gql/generated/types";
+import { UPDATE_SPAWN_HOST_STATUS } from "gql/mutations";
+import { GET_MY_HOSTS } from "gql/queries";
+import { useNetworkStatus } from "hooks";
 import { HostStatus } from "types/host";
 import { MyHost } from "types/spawn";
 
 export const SpawnHostActionButton: React.FC<{ host: MyHost }> = ({ host }) => {
+  const dispatchBanner = useBannerDispatchContext();
+
+  const glyph = mapStatusToGlyph[host.status];
   const action = mapStatusToAction[host.status];
 
-  return action ? (
-    <PaddedButton glyph={<Icon glyph={action} />} size={Size.XSmall} />
-  ) : null;
+  // When the UPDATE_SPAWN_HOST_STATUS mutation occurs the host state is not immediately updated, It gets updated a few seconds later.
+  // Since the GET_MY_HOSTS query on this components parent polls at a slower rate, this component triggers a poll at a faster interval for that
+  // query when it returns an updated host status the polling is halted. This allows the query to poll slowly and not utilize unnecessary bandwith
+  // except when an action is performed and we need to fetch updated data.
+  const [getMyHosts, { startPolling, stopPolling }] = useLazyQuery<
+    MyHostsQuery,
+    MyHostsQueryVariables
+  >(GET_MY_HOSTS, {
+    pollInterval: 1000,
+    onError: (e) => {
+      dispatchBanner.errorBanner(
+        `There was an error loading your spawn hosts: ${e.message}`
+      );
+    },
+  });
+  useNetworkStatus(startPolling, stopPolling);
+
+  // Stop polling when we get updated host data
+  useEffect(() => {
+    if (stopPolling) {
+      stopPolling();
+    }
+  }, [host]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
this seems to work because the MyHost query in SpawnHost.tsx only creates a new reference to data and data.myHosts.hosts when the data fields change. 
Do you think it makes sense diff `host.id` and `host.status` instead of the entire host object so we can be more specific about when to stop polling? 
This could save us from an edgecase where we stop polling prematurely after we update two host states and stop polling for both hosts as soon as the references inside the `data.myHost.hosts` change once.

Reply:
I dont think this can happen. When we render each row in the spawn host table we use the host id as the row key. By doing this react knows to update only the row with the matching host. So there shouldn't be an issue from host ids changing and effecting other rows and causing other components props to change.

 We pass the individual host into each spawn Host action button and each one performs its own query and has its own hook to manage polling. Therefore the stopPolling call of one hook would not affect another component.